### ISSUE TITLE
Import-tool: Make import API run channel imports in parallel

### ISSUE THREAD

**Issue by PascalPrecht** [status-im/status-desktop#6692]
The current implementation processes each channel file to import in a serial fashion. 
Because there are many tasks that need to be done on a per channel basis, it makes sense to paralleize the process.

Also, channel imports could run into issues individually, so it's probably better to also error out of an individual thread instead of the main thread.

Implementing this also means that we need to extend existing signals and introduce new ones:

- Channel import started
- Channel import failed
- Channel import sucess/finished
- Total import done signal

Obviously, when failed, we need to emit the error accordingly.

**Comment by PascalPrecht** [comment]
Played around with this, however, turns out that this introduces some funny unexpected results. Probably because everytime we create a channel, we update the community description. If multiple threads try to do this at the same time, they'll have different descriptions individually, resulting in missing channels.

I'm going to put this on hold for now to revisit this later. Will be doing this operation non-parallel for now.

**Comment by PascalPrecht** [comment]
This is an optimization that is not crucial for 0.8.0 /cc @iurimatias

**Comment by jrainville** [comment]
Closing for now since it doesn't seem that necessary for now.

Pascal: "The channels creation itself and importing messages doesn't seem to perform badly. I doubt making this run in parallel is worth the effort. It'll introduce a lot of complexity with very little upside.

This issue was created at a time where there were many unknowns."
